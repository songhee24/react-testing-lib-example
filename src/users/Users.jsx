import React from "react";
import axios from "axios";

const Users = () => {
  const [users, setUsers] = React.useState([]);

  const loadUsers = async () => {
    const resp = await axios.get("https://jsonplaceholder.typicode.com/users");
    setUsers(resp.data);
  };

  React.useEffect(() => {
    loadUsers();
  }, []);

  return (
    <div>
      {users.map((user) => (
        <div key={user.id} data-testid="user-item">
          {user.name}
        </div>
      ))}
    </div>
  );
};

export default Users;
