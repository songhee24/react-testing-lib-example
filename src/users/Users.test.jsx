import { fireEvent, render, screen } from "@testing-library/react";
import Users from "./Users";
import axios from "axios";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import UserDetailsPage from "../pages/UserDetailsPage";
import React from "react";
jest.mock("axios");
describe("USERS TEST", () => {
  let response;

  beforeEach(() => {
    response = {
      data: [
        {
          id: 1,
          name: "Leanne Graham",
        },
        {
          id: 2,
          name: "Ervin Howell",
        },
        {
          id: 3,
          name: "Clementine Bauch",
        },
      ],
    };
  });

  test("renders users test", async () => {
    axios.get.mockReturnValue(response);
    render(<Users />);
    const users = await screen.findAllByTestId("user-item");
    expect(users.length).toBe(3);
    expect(axios.get).toBeCalledTimes(1);
    screen.debug();
    expect(users).toMatchSnapshot();
  });

  test("redirect to details page test", async () => {
    axios.get.mockReturnValue(response);
    render(
      <MemoryRouter initialEntries={["/users"]}>
        <Routes>
          <Route path="/users" element={<Users />} />
          <Route path="/users/:id" element={<UserDetailsPage />} />
        </Routes>
      </MemoryRouter>
    );
    const users = await screen.findAllByTestId("user-item");
    expect(users.length).toBe(3);
    fireEvent.click(users[0]);
    expect(screen.getByTestId("user-page")).toBeInTheDocument();
  });
});
