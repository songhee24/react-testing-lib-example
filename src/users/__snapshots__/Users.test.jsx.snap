// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`USERS TEST renders users 1`] = `
Array [
  <div
    data-testid="user-item"
  >
    Leanne Graham
  </div>,
  <div
    data-testid="user-item"
  >
    Ervin Howell
  </div>,
  <div
    data-testid="user-item"
  >
    Clementine Bauch
  </div>,
]
`;
